### ISSUE TITLE
Make all DynamicTables chunked and resizable by default

### ISSUE THREAD

**Issue by rly** [NeurodataWithoutBorders/pynwb#1067]
## Feature Request

It has often been requested that a table can be appended to, e.g. the units table, epochs table, invalid time intervals tables, etc. over the course of data processing and analysis. In order to allow this, DynamicTables need to be chunked and resizable by default. See also #918 and #920 and #1059

**Comment by rly** [comment]
@ajtritt now that `DynamicTable` is in HDMF, how should we go about making `DynamicTable` chunked and resizable by default in PyNWB and not in HDMF as you had suggested? Do we need to subclass DynamicTable or is there a better way?

**Comment by ajtritt** [comment]
> Do we need to subclass DynamicTable or is there a better way?

No, you would need to subclass [DynamicTableMap](https://hdmf.readthedocs.io/en/latest/hdmf.common.io.table.html#hdmf.common.io.table.DynamicTableMap) and extend [get_attr_value](https://hdmf.readthedocs.io/en/latest/hdmf.build.map.html#hdmf.build.map.ObjectMapper.get_attr_value)

**Comment by stephprince** [comment]
related: https://github.com/hdmf-dev/hdmf/pull/1093

**Comment by bendichter** [comment]
@rly can we close?
